### PR TITLE
Fix session load state & run extra_postpasses only once

### PR DESCRIPTION
**Pull request by BowenBao**
* `load_state_dict` is now fixed to also load training state as well.
* remove redundant code calling extra_postpasses twice.

### CODE REVIEW

File: orttraining/orttraining/test/python/onnxruntime_test_postprocess.py [hunk 1]
```diff
@@ -209,6 +209,70 @@ def _bert_helper(self, onnx_model):
         assert model_info[0].name == expand_nodes[0].output[0]
         assert model_info[0].type == onnx_model.graph.input[0].type
 
+    def test_extra_postpass(self):
+        def postpass_replace_first_add_with_sub(model):
```
Comment (thiagocrepaldi) [inline orttraining/orttraining/test/python/onnxruntime_test_postprocess.py:213]: Should we test both cases where the model is direct ONNX model and when it is a torch.nn.Module being converted into ONNX?

**Comment by ytaous** [comment]
During some tests, I am encountering an issue with transformers pkg. I found the latest 2.8.0 is not compatible, and it works if I reinstall it with 2.0.0. 
Should we doc it somewhere for 1p/3p reference? 

"/bert_data/thiagofc/dev/devops/ELR/sources/dev/SubstrateInferences/ELR/src/elr_pretrain/model/turing.py", line 8, in __init__
    self.bert = BertModel(config)
  File "/bert_ort/ettao/onnx/ettao_py36/lib/python3.6/site-packages/transformers/modeling_bert.py", line 615, in __init__
    self.embeddings = BertEmbeddings(config)
  File "/bert_ort/ettao/onnx/ettao_py36/lib/python3.6/site-packages/transformers/modeling_bert.py", line 150, in __init__
    self.word_embeddings = nn.Embedding(config.vocab_size, config.hidden_size, padding_idx=config.pad_token_id)
  File "/bert_ort/ettao/onnx/ettao_py36/lib/python3.6/site-packages/torch/nn/modules/sparse.py", line 100, in __init__
    self.weight = Parameter(torch.Tensor(num_embeddings, embedding_dim))
TypeError: new() received an invalid combination of arguments - got (str, int), but expected one of:
 * (*, torch.device device)

**Comment by ytaous** [comment]
@thiago@thiagocrepaldi.com 

---
In reply to: [645095692](https://github.com/microsoft/onnxruntime/pull/4255#issuecomment-645095692) [](ancestors = 645095692)

**Comment by ytaous** [comment]
@thiago@thiagocrepaldi.com 

---
In reply to: [645095692](https://github.com/microsoft/onnxruntime/pull/4255#issuecomment-645095692) [](ancestors = 645095692)

**Comment by thiagocrepaldi** [comment]
> 
> 
> During some tests, I am encountering an issue with transformers pkg. I found the latest 2.8.0 is not compatible, and it works if I reinstall it with 2.0.0.
> Should we doc it somewhere for 1p/3p reference?
> 
> "/bert_data/thiagofc/dev/devops/ELR/sources/dev/SubstrateInferences/ELR/src/elr_pretrain/model/turing.py", line 8, in **init**
> self.bert = BertModel(config)
> File "/bert_ort/ettao/onnx/ettao_py36/lib/python3.6/site-packages/transformers/modeling_bert.py", line 615, in **init**
> self.embeddings = BertEmbeddings(config)
> File "/bert_ort/ettao/onnx/ettao_py36/lib/python3.6/site-packages/transformers/modeling_bert.py", line 150, in **init**
> self.word_embeddings = nn.Embedding(config.vocab_size, config.hidden_size, padding_idx=config.pad_token_id)
> File "/bert_ort/ettao/onnx/ettao_py36/lib/python3.6/site-packages/torch/nn/modules/sparse.py", line 100, in **init**
> self.weight = Parameter(torch.Tensor(num_embeddings, embedding_dim))
> TypeError: new() received an invalid combination of arguments - got (str, int), but expected one of:
> 
>     * (*, torch.device device)

Isn't this a constraint on ELR model as opposed to ORT?

**Comment by ytaous** [comment]
I thought this is general issue, as I am getting the same error from testing UNILM. 
Close it as long as you are aware of it :-)

---
In reply to: [645487592](https://github.com/microsoft/onnxruntime/pull/4255#issuecomment-645487592) [](ancestors = 645487592)

**Pull request merged**